internal/provider/incident_catalog_entries_resource.go

```diff
@@ -600,3 +622,58 @@ func (r *IncidentCatalogEntriesResource) reconcile(ctx context.Context, data *In
 
 	return catalogType, entries, nil
 }
+
+// isAttributeManaged checks if the given attribute should be managed by this resource.
+func (m *IncidentCatalogEntriesResourceModel) isAttributeManaged(attributeID string) bool {
+	// Check if the attribute is in the managed list
+	attrSet, known := m.managedAttributesSet()
+	if !known {
+		return true
+	}
+
+	return attrSet[attributeID]
+}
+
+func (m *IncidentCatalogEntriesResourceModel) managedAttributesSet() (map[string]bool, bool) {
+	if m.ManagedAttributes.IsNull() || m.ManagedAttributes.IsUnknown() {
+		return nil, false
+	}
+
+	if m.managedAttrSet != nil {
+		return m.managedAttrSet, true
+	}
+
+	managedAttrSet := map[string]bool{}
+	for _, attrElem := range m.ManagedAttributes.Elements() {
+		if attrElem.IsUnknown() {
+			return nil, false
+		}
+
+		attrIDStr, ok := attrElem.(types.String)
+		if !ok {
+			// Weird but ok
+			return nil, false
+		}
+
+		managedAttrSet[attrIDStr.ValueString()] = true
+	}
+
+	// Technically this isn't race-safe, since multiple goroutines _could_ try to set this at once.
+	// However, it's a static value that will be the same however it's built, so not worried about this right now.
+	m.managedAttrSet = managedAttrSet
+	return managedAttrSet, true
+}
+
+func (m *IncidentCatalogEntriesResourceModel) buildUpdateAttributes() *[]string {
+	if m.ManagedAttributes.IsNull() {
```

Comment:
|| m.ManagedAttributes.IsUnknown()? should we have a helper for this, since we use it twice?

Reply:
Same as the comment above!